### PR TITLE
additional changes to met workflow functions

### PR DESCRIPTION
**Pull request by bcow**
- new read.register function for checking register.met.xml for missing values and filling in what it can
- various small fixes of typos etc
- cleaning up print statements and changing some of them over to logger.info - should discuss what's still being just printed (for example, all the checks) - whether we want it as part of the output or not.
- made a small change to metgapfill to prevent files that aren't completely gapfilled from being left in the met_CF_gapfill_* folders.

I've tested all the new changed on https://test-pecan.bu.edu/pecan/ which only has SIPNET so quick test runs should be done with the other three models on pecan2 - but they were working and these changes hopefully shouldn't change whether or not the models run.

### CODE REVIEW

File: modules/data.atmosphere/R/read.register.R [hunk 1]
```diff
@@ -0,0 +1,44 @@
+
+##' @name read.register
+##' @title read.register
+##' @export
+##' @param register.xml path of xml file
+##' @param con betydb connection
+##' 
+##' @author Betsy Cowdery
```
Comment (robkooper) [inline modules/data.atmosphere/R/read.register.R]: Can you move this down to the function, and move the library statements up.

File: modules/data.atmosphere/R/metgapfill.R [hunk 2]
```diff
@@ -335,6 +335,7 @@ metgapfill <- function(in.path, in.prefix, outfolder, start_date, end_date, lst=
     nc_close(nc)
     
     if (length(error) > 0) {
+      file.remove(new.file)
```
Comment (robkooper) [inline modules/data.atmosphere/R/metgapfill.R]: If you remove the file, the resuls will not be there. Maybe rename the file to paste(new.file, "bad", sep=".")

File: modules/data.atmosphere/R/read.register.R (hunk 1 above)
Comment (bcow) [inline modules/data.atmosphere/R/read.register.R]: Done

File: modules/data.atmosphere/R/metgapfill.R (hunk 2 above)
Comment (bcow) [inline modules/data.atmosphere/R/metgapfill.R]: That makes sense - I changed the code to do that.

**Pull request merged**